include/intx/intx.hpp

```diff
@@ -70,6 +61,17 @@
 
 namespace intx
 {
+/// Mark a possible code path as unreachable (invokes undefined behavior).
+/// TODO(C++23): Use std::unreachable().
+[[noreturn]] inline void unreachable() noexcept
+{
+#if __has_builtin(__builtin_unreachable)
+    __builtin_unreachable();
+#elif defined(_MSC_VER)
+    __assume(false);
+#endif
+}
```

Comment:
What happens in the else case? Do we use this mostly for optimisations or also expect it as a safeguard?

Reply:
Mostly for optimization but sometimes helps with compiler warnings. Undefined Behavior Sanitizer has it instrumented.